### DISCUSSION TITLE
Feature Request: tpch,tpcds support generate data to parquet

### DISCUSSION THREAD

**Discussion by digoal** [duckdb/duckdb#9858] [Ideas]
Why do you want this feature?

Most of the time, our data is stored in Parquet files. However, the current data generation for TPC-H and TPC-DS only supports in-memory storage and does not directly generate Parquet files. This can easily lead to OOM (out-of-memory) issues and make it inconvenient to test large datasets.
```
D call dbgen(sf='10'); 
Killed
```

We hope to add an option to the TPC-H and TPC-DS extensions that allows data generation to be in either Parquet format or in-memory storage.

best regards,
digoal

**Comment by carlopi** [comment]
Can you try:
```
$ duckdb
D ATTACH 'tpch.db';
D USE tpch;
D call dbgen(sf='10');
```
or the equivalent
```
$ ducdkb tpch.db
D call dbgen(sf='10');
```

This should generate a tpch.db DuckDB-database with the relevant data using a storage-backed database.

```
$ duckdb tpch.db
D #... do queries
```

Basically, and if this is not obvious to you we should document this better, there are two main modes:

`$ duckdb [filename]`
uses by default a storage-backed DB and persist data there

`$ duckdb`
uses a non-persistent database

This is the same in all APIs, but syntax will be different in how to specify this option.

**Comment by l1t1** [comment]
you can copy to parquet by
```
select 'copy '|| table_name||' to '''||table_name ||'.parquet'';' from duckdb_tables;
```
